### PR TITLE
Added hls support for media window

### PR DESCRIPTION
**Pull request by CandyFace**
Added hls.js dependency to package and hls support for window.
Scaling for hls is not tested thoroughly because i only had one tweet with such content. Also for some reason I wasn't able to get the actual video size of the video container for the new format but a downscaled value instead, which means that I'm multiplying manually. Whether this works for all cases, I don't know, so if you experience weird sizing with such content, that might be the issue.

### CODE REVIEW

**Comment by CandyFace** [comment]
Alright all the changes that I've made now should fix all the problems we've discussed.
In addition I also made it possible to drag a window now.

I believe that the change from 'on' to 'once' fixed the issue that windows would be rescaled based on the new popup window.

File: src/containers/tweet-container.js [hunk 1]
```diff
@@ -39,14 +37,11 @@ export default class TweetContainer extends BaseContainer {
...
-    ipcMain.on('imageDimensions', (event, width, height) => {
+    ipcMain.once('imageDimensions', (event, width, height) => {
       if (win != null) {
         const screenWidth  = Math.round((screen.width/2) - (width/2));
         const screenHeight = Math.round((screen.height/3) - (height/3));
-        let options = { x: screenWidth, y: screenHeight, width: width, height: height };
-        if (process.platform === 'darwin') {
-          Object.assign(options, { height: height+20 });
-        }
+        let options = { x: screenWidth, y: screenHeight, width: width, height: height+20 };
```
Comment (k0kubun) [inline src/containers/tweet-container.js]: Why did you change this? I guess you don't consider Linux environment.

File: src/media-popup.js [hunk 2]
```diff
@@ -1,21 +1,33 @@
 // FIXME: Refactor later
-var ipcRenderer = require('electron').ipcRenderer;
+import { ipcRenderer }        from 'electron';
```
Comment (k0kubun) [inline src/media-popup.js]: Too many spaces.

File: src/containers/tweet-container.js (hunk 1 above)
Comment (k0kubun) [inline src/containers/tweet-container.js]: Ah, does `body.style = "padding-top: 0px;"` do it?

File: src/media-popup.js [hunk 3]
```diff
@@ -1,21 +1,33 @@
...
 var img = document.getElementById('loadedImage');
 var video = document.getElementById('loadedVideo');
+var body = document.getElementsByTagName("body")[0];
+var hlsMedia = new hls();
 var mediaWidth = 0,
 mediaHeight = 0,
 maxWidth = 800,
 maxHeight = 600;
 
+if (process.platform !== 'darwin'){
```
Comment (k0kubun) [inline src/media-popup.js]: Your coding style isn't unified. Please put a space before `{` every time.

File: src/media-popup.js [hunk 4]
```diff
@@ -1,21 +1,33 @@
...
     mediaHeight = this.height;
   } else {
-    mediaWidth = this.videoWidth;
-    mediaHeight = this.videoHeight;
+    if (video.title === "HLS media"){
+      mediaWidth = this.videoWidth * 4;
+      mediaHeight = this.videoHeight * 4;
+    } else {
+      mediaWidth = this.videoWidth
+      mediaHeight = this.videoHeight
```
Comment (k0kubun) [inline src/media-popup.js]: Why do you skip `;`? 
This is not CoffeeScript but JavaScript and I use conservative style.

File: src/media-popup.js [hunk 5]
```diff
@@ -42,16 +54,22 @@ function resizeMedia(media) {
...
-    video.src = mediaUrl;
+    video.src = mediaUrl
     video.onloadedmetadata = resizeMedia.bind(video);
     video.preload = "auto";
     video.loop = true;
     video.autoplay = true;
     video.playbackRate = 1;
   }
 });
+
```
Comment (k0kubun) [inline src/media-popup.js]: Unnecessary blank line.

File: src/containers/tweet-container.js (hunk 1 above)
Comment (CandyFace) [inline src/containers/tweet-container.js]: Yes exactly, I tested the height and it should always add 20 to the height, else the image will be cropped. To adjust for the title on Linux and Windows, the padding is set to zero on the body and only +20 on mac where we have hidden titlebar

File: src/media-popup.js (hunk 2 above)
Comment (CandyFace) [inline src/media-popup.js]: Will change again, thought i wanted to do it like you have in your other files in the top but nvm. will change again.

File: src/media-popup.js (hunk 4 above)
Comment (CandyFace) [inline src/media-popup.js]: Hmm ... good question I just don't think about it in Javascript since the compiler doesn't seem to care about it. I usually do it, just not in Javascript. Will fix.

File: src/media-popup.js [hunk 6]
```diff
@@ -54,4 +71,4 @@ ipcRenderer.on('load-media', (event, mediaUrl, mediaType) => {
     video.autoplay = true;
     video.playbackRate = 1;
   }
-});
+});
```
Comment (k0kubun) [inline src/media-popup.js]: While blank line isn't recommended, a newline character should be put at the end of the file.
(`});` is bad, `});\n` is good and `});\n\n` is bad too.)

Comment (CandyFace) [inline src/media-popup.js]: Interesting, is there a reason for that, or is that just because that's how you style your documents?

Comment (k0kubun) [inline src/media-popup.js]: > `});` is bad

This is posix rule.
http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_206
http://stackoverflow.com/questions/5813311/no-newline-at-end-of-file

> `});\n\n` is bad too

This is my style. Isn't it just lengthy?

**Comment by k0kubun** [comment]
Everything works absolutely fine on both macOS and Linux. Great job!!

In the future, I hope that I will become able to open and close this popup by shortcut key. Maybe <kbd>e</kbd> (like a client I use formerly) and <kbd>Esc</kbd>?

**Comment by CandyFace** [comment]
Yay╭( ･ㅂ･)و ̑̑ ˂ᵒ͜͡ᵏᵎ⁾✩
Hmm yes shortcuts would be nice, as would a window with actual preferences.
